File: javascript/high-scores/high-scores.js
export class HighScores {
  constructor(input) {
   this._scores = input;
  }

  get scores(){
    return this._scores;
  }

  get latest() {
    return Number(this._scores.slice(-1));
  }

  get personalBest() {
    return Math.max(...this._scores);
  }

  get personalTopThree() {
    let result = this._scores;
    return result.sort((x, y) => y - x).slice(0, 3);
  }
}

File: csharp/high-scores/HighScores.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class HighScores
{
    public List<int> _highScores;
    public HighScores(List<int> list) => _highScores = list;

    public List<int> Scores() => _highScores.ToList();
    public int Latest() => _highScores.LastOrDefault();
    public int PersonalBest() => _highScores.Max();


    public List<int> PersonalTopThree() =>
         _highScores.OrderByDescending(x => x).Take(3).ToList();
}